### ISSUE TITLE
IO::Scalar and bleadperl

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5470]
Migrated from [rt.perl.org#9368](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=9368) (status was 'resolved')

Searchable as RT9368$

**Comment by p5pRT** [comment]
### From acme@dhcp-client-13.dev-fotango.com

#### Created by acme@dhcp-client-13.dev-fotango.com



I develop using bleadperl\[1\] and for the past couple of days bleadperl and
IO&#8203;::Scalar haven't been playing nicely\. I've done a quick search but
nothing popped out\. To be honest\, I have no idea what the problem is\,
so here is a quick heads up\.

\[If this is a manifestation of some other bug that is currently being 
discussed\, great\! I'll look like an idiot ;\-\] 

\[acme@&#8203;dhcp\-client\-13 IO\-stringy\-2\.108\]$ make test
PERL\_DL\_NONLAZY=1 /home/acme/bin/perl5\.7\.3 "\-MExtUtils&#8203;::Command&#8203;::MM" "\-e" "test\_harness\(0\, 'blib/lib'\, 'blib/arch'\)" t/\*\.t
t/IO\_Lines\.\.\.\.\.\.\.\.\.\.ok                                                       
t/IO\_Scalar\.\.\.\.\.\.\.\.\.ok                                                       
t/IO\_ScalarArray\.\.\.\.ok                                                       
t/IO\_WrapTie\.\.\.\.\.\.\.\.ok                                                       
t/simple\.\.\.\.\.\.\.\.\.\.\.\.Can't call method "PRINT" on an undefined value at t/simple\.t line 29\.
t/simple\.\.\.\.\.\.\.\.\.\.\.\.dubious                                                  
&nbsp;    Test returned status 255 \(wstat 65280\, 0xff00\)
DIED\. FAILED tests 1\-6
&nbsp;    Failed 6/6 tests\, 0\.00% okay
t/two\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.Can't call method "PRINT" on an undefined value at t/two\.t line 36\.
t/two\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.dubious                                                  
&nbsp;    Test returned status 255 \(wstat 65280\, 0xff00\)
DIED\. FAILED tests 1\-2
&nbsp;    Failed 2/2 tests\, 0\.00% okay
Failed Test Stat Wstat Total Fail  Failed  List of Failed
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
t/simple\.t   255 65280     6    6 100\.00%  1\-6
t/two\.t      255 65280     2    2 100\.00%  1\-2
Failed 2/6 test scripts\, 66\.67% okay\. 8/116 subtests failed\, 93\.10% okay\.
make&#8203;: \*\*\* \[test\_dynamic\] Error 2

Cheers\, Leon

\[1\] See\, aren't I nice \- I try to find bugs for you guys ;\-\)

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.7.3:

Configured by acme at Wed May 15 09:41:06 BST 2002.

Summary of my perl5 (revision 5.0 version 7 subversion 3 patch 16606) configuration:
  Platform:
    osname=linux, osvers=2.4.2-2, archname=i686-linux
    uname='linux dhcp-client-13.dev-fotango.com 2.4.2-2 #1 sun apr 8 20:41:30 edt 2001 i686 unknown '
    config_args='-ds -e -Dusedevel -Dprefix=/home/acme'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O2',
    cppflags='-fno-strict-aliasing -I/usr/local/include -I/usr/include/gdbm'
    ccversion='', gccversion='2.96 20000731 (Red Hat Linux 7.1 2.96-85)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldl -lm -lc -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lc -lcrypt -lutil
    libc=/lib/libc-2.2.4.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    DEVEL16572


@INC for perl v5.7.3:
    /home/acme/lib/perl5/5.7.3/i686-linux
    /home/acme/lib/perl5/5.7.3
    /home/acme/lib/perl5/site_perl/5.7.3/i686-linux
    /home/acme/lib/perl5/site_perl/5.7.3
    /home/acme/lib/perl5/site_perl
    .


Environment for perl v5.7.3:
    HOME=/home/acme
    LANG=en_GB
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/acme/bin:/home/acme/bin:/home/acme/bin:/bin:/usr/bin:/usr/bin/X11:/usr/local/bin:/usr/games:/opt/gnome/bin:/usr/X11R6/bin:/home/acme/bin
    PERL_BADLANG (unset)
    SHELL=/bin/bash



```
</details>

**Comment by p5pRT** [comment]
### From @ysth

I think I broke this with \#16501\.  Nick fixed it with \#16810\.  Both
were retracted with \#16485\.  Original problem that prompted \#16501
now fixed with \#18121\.

**Comment by p5pRT** [comment]
@ysth - Status changed from 'open' to 'resolved'
